prime-router/src/main/kotlin/azure/HistoryFunctions.kt

```diff
@@ -332,6 +345,8 @@ open class BaseHistoryFunction : Logging {
                 .build()
         } catch (ex: Exception) {
             context.logger.info("Exception during creating of reports list - file not found")
+            context.logger.severe(ex.message)
+            context.logger.severe(ex.stackTraceToString())
             response = request.createResponseBuilder(HttpStatus.NOT_FOUND)
```

Comment:
This is a generic Exception - are we sure its _always_ a file not found issue?   You should always log the name of the receiving org - that's super useful in practice.   And the filename of course if it exists - but maybe that's in the ex.message?

And will this code even ever execute?   Its very hard to tell what code might throw an Exception , but the most likely seems like it would be `workflowEngine.fetchHeader` line above,  which already swallows its exception.